Sync/SyncStateMachine.swift

```diff
@@ -73,10 +73,20 @@ open class SyncStateMachine {
     var stateLabelsSeen = [SyncStateLabel: Bool]()
     var stateLabelSequence = [SyncStateLabel]()
 
+    let stateLabelsIncluded: Set<SyncStateLabel>
+
     let scratchpadPrefs: Prefs
 
-    public init(prefs: Prefs) {
+    public convenience init(prefs: Prefs, optimistic: Bool = false) {
```

Comment:
Added `init` to allow

```swift
SyncStateMachine(prefs: prefs, allowingStates: SyncStateMachine.OptimisticStates)
```

Removed Bool flag. I'm not usually a fan of flags, not sure why I did this.